src/csharp/Microsoft.Spark/Sql/ArrowArrayHelpers.cs

```diff
@@ -38,110 +39,84 @@ internal static class ArrowArrayHelpers
             ArrowTypeId.Binary,
         };
 
-        public static IArrowArray CreateEmptyArray<T>()
+        public static BaseColumn CreateEmptyColumn<T>()
         {
-            ArrayData data = BuildEmptyArrayDataFromArrayType<T>();
-            return ArrowArrayFactory.BuildArray(data);
-        }
-
-        public static IArrowArray CreateEmptyArray(IArrowType arrowType)
-        {
-            ArrayData data = BuildEmptyArrayDataFromArrowType(arrowType);
-            return ArrowArrayFactory.BuildArray(data);
-        }
-
-        private static ArrayData BuildEmptyArrayDataFromArrowType(IArrowType arrowType)
-        {
-            if (s_twoBufferArrowTypes.Contains(arrowType.TypeId))
+            BaseColumn ret;
+            if (typeof(T) == typeof(PrimitiveColumn<bool>))
             {
-                return new ArrayData(arrowType, 0,
-                    buffers: new[] { ArrowBuffer.Empty, ArrowBuffer.Empty });
+                ret = new PrimitiveColumn<bool>("Empty");
```

Comment:
Is the `"Empty"` column name something that will be checked in the future to validate that this designates an empty column ?  If so should we pull this name out and define it as some kind of constant ?

Reply:
No, `Empty` is not a special name. I just put it here to make it readable. Maybe I should change it to "Return" to avoid confusion